linera-views/src/key_value_store_view.rs

```diff
@@ -190,6 +190,26 @@ where
         *self.hash.get_mut() = self.stored_hash;
     }
 
+    async fn has_pending_changes(&self) -> bool {
+        if self.delete_storage_first {
+            return true;
+        }
+        if !self.deleted_prefixes.is_empty() {
+            return true;
+        }
+        if !self.updates.is_empty() {
+            return true;
+        }
+        if self.sizes.has_pending_changes().await {
+            return true;
+        }
+        let hash = self.hash.lock().await;
+        if self.stored_hash != *hash {
+            return true;
+        }
+        self.stored_total_size != self.total_size
```

Comment:
Optional nit: it might make sense to do all `sync` checks before the `async` checks, just in case they are faster.

Reply:
Yes, you are right.